### PR TITLE
Geom_Point Fix

### PR DESCRIPTION
**Pull request by HuntJSparra**
Fixed issue mentioned in #4424 

Previously converting mpl Paths or PathCollections to Bokeh glyphs was not supported. Unfortunately, mpl.gcf() for whatever reason returns a PathCollection instead of a Line2D when handling geom_point from ggplot. This caused the geom_point-turned-PathCollection not to render on the Bokeh plot. Shamelessly copying the draw_markers function from bokeh_renderer into a new draw_path_collection and modifying it a bit to work allows for the paths to be turned into Bokeh glyphs.

Unfortunately there are three limitations:
1. Since the PathCollection stores the shape of the glyph as a path and Bokeh does not yet handle custom glyph shapes, all converted paths default to circle.
2. If someone were to pass a PathCollection for a type of graph other than a scatter plot, it would most likely (untested) result in a scatter plot of the major points instead of, for example, a line plot. Although, this is still better than drawing nothing at all imo.
3. My method of calculating the sizes is no where close to an exact science.

Before Fix:
3c419be0-28d2-11e6-8161-0b7964395ab0

After Fix:
ggplot_works_mostly

### CODE REVIEW

**Comment by bryevdv** [comment]
@HuntJSparra this looks good, can you do two more things:

* add the code that generated the above image to a new script in `examples/compat` 
* run all the other `examples/compat` scripts just to make sure this doesn't add any unexpected things to existing examples there

**Comment by HuntJSparra** [comment]
- Added Example
- Added support for multiple colors on the same path
- Added support for path sizes although it is not an exact conversion
- Tested and works with current examples

Some of the new code is very... unpythonic. If someone could help clean it up I would really appreciate it.

File: examples/compat/ggplot_point.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from bokeh import mpl
+from bokeh.plotting import output_file, show
+
+g = ggplot(mtcars, aes(x='wt', y='mpg', color='qsec')) + geom_point()
+g.make()
+import pdb; pdb.set_trace()
+plt.title("Point ggplot-based plot in Bokeh.")
+
+output_file("ggplot_point.html", title="ggplot_point.py example")
+#plt.show()
```
Comment (bryevdv) [inline examples/compat/ggplot_point.py]: would remove commented line

**Comment by bryevdv** [comment]
LGTM, will merge when tests are green

**Pull request merged**